### PR TITLE
Sanitize Error When invalid MSI ID is provided

### LINKED ISSUE #7369: Attempting login with a bogus managed identity ID crashes
**Describe the bug**

Attempting to `az login` using a bogus managed identity ID leads to a crash instead of any informative error message:

```
# python3 -m azure.cli login --identity -u beep-boop-doop

Traceback (most recent call last):
  File "azure/cli/core/_profile.py", line 277, in find_subscriptions_in_vm_with_msi
    msi_creds = MSIAuthentication(resource=resource, client_id=identity_id)
  File "msrestazure/azure_active_directory.py", line 576, in __init__
    self.set_token()
  File "msrestazure/azure_active_directory.py", line 584, in set_token
    token_entry = self._vm_msi.get_token(self.resource)
  File "msrestazure/azure_active_directory.py", line 632, in get_token
    token_entry = self._retrieve_token_from_imds_with_retry(resource)
  File "msrestazure/azure_active_directory.py", line 669, in _retrieve_token_from_imds_with_retry
    raise HTTPError(request=result.request, response=result.raw)
requests.exceptions.HTTPError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "knack/cli.py", line 197, in invoke
    cmd_result = self.invocation.execute(args)
  File "azure/cli/core/commands/__init__.py", line 369, in execute
    six.reraise(*sys.exc_info())
  File "six.py", line 693, in reraise
    raise value
  File "azure/cli/core/commands/__init__.py", line 343, in execute
    result = cmd(params)
  File "azure/cli/core/commands/__init__.py", line 182, in __call__
    return self.handler(*args, **kwargs)
  File "azure/cli/core/__init__.py", line 436, in default_command_handler
    result = op(**command_args)
  File "azure/cli/command_modules/profile/custom.py", line 107, in login
    return profile.find_subscriptions_in_vm_with_msi(username)
  File "azure/cli/core/_profile.py", line 282, in find_subscriptions_in_vm_with_msi
    msi_creds = MSIAuthentication(resource=resource, object_id=identity_id)
  File "msrestazure/azure_active_directory.py", line 576, in __init__
    self.set_token()
  File "msrestazure/azure_active_directory.py", line 584, in set_token
    token_entry = self._vm_msi.get_token(self.resource)
  File "msrestazure/azure_active_directory.py", line 632, in get_token
    token_entry = self._retrieve_token_from_imds_with_retry(resource)
  File "msrestazure/azure_active_directory.py", line 669, in _retrieve_token_from_imds_with_retry
    raise HTTPError(request=result.request, response=result.raw)
requests.exceptions.HTTPError
root@clogs-gouda-vfmijq:~#
```

**Expected behavior**

An informative error message of _some_ sort, not just an angry red traceback.

**Environment summary**

Ubuntu 18.04, pip, azure-cli 2.0.45, bash

**Additional context**

Adding `--debug` reveals that the tool attempts these two requests to the Azure Instance Metadata Service:

* http://169.254.169.254/metadata/identity/oauth2/token?resource=https%3A%2F%2Fmanagement.core.windows.net%2F&api-version=2018-02-01&client_id=beep-boop-doop
* http://169.254.169.254/metadata/identity/oauth2/token?resource=https%3A%2F%2Fmanagement.core.windows.net%2F&api-version=2018-02-01&object_id=beep-boop-doop

### PR DESCRIPTION
**Pull request by marstr**
Fixes #7369 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -283,15 +283,31 @@ def find_subscriptions_in_vm_with_msi(self, identity_id=None, allow_no_subscript
...
                         raise
+
+                if not authenticated:
+                    try:
+                        identity_type = MsiAccountTypes.user_assigned_object_id
+                        msi_creds = MSIAuthentication(resource=resource, object_id=identity_id)
+                        authenticated = True
+                    except HTTPError as ex:
+                        if ex.response.reason == 'Bad Request' and ex.response.status == 400:
+                            logger.info('failed to use MSI as user assigned resource id')
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/_profile.py]: suggest `Not an object id of MSI`

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -283,15 +283,31 @@ def find_subscriptions_in_vm_with_msi(self, identity_id=None, allow_no_subscript
...
+                authenticated = False
                 try:
                     msi_creds = MSIAuthentication(resource=resource, client_id=identity_id)
                     identity_type = MsiAccountTypes.user_assigned_client_id
+                    authenticated = True
                 except HTTPError as ex:
                     if ex.response.reason == 'Bad Request' and ex.response.status == 400:
-                        identity_type = MsiAccountTypes.user_assigned_object_id
-                        msi_creds = MSIAuthentication(resource=resource, object_id=identity_id)
+                        logger.info('failed to use MSI as user assigned client id')
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/_profile.py]: suggest `Not a client id of MSI`

Comment (marstr) [reply src/azure-cli-core/azure/cli/core/_profile.py]: Changed in 9b9b061

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 1 above)
Comment (marstr) [reply src/azure-cli-core/azure/cli/core/_profile.py]: Changed in 9b9b061

**Comment by marstr** [comment]
Looks like CI is failing because of missing resources during the test run. I'm going to go ahead and assume that this is one of two possible problems:
1. Just temporary trouble that will pass naturally.
1. Recording that need to be updated (and probably are on the dev branch.)

I'm going to rebase and let CI run again.

**Pull request merged**